app/javascript/packs/TagsIndex.jsx

```diff
@@ -0,0 +1,90 @@
+import { h, render } from 'preact';
```

Comment:
This is in the description but dropping it here in case its missed.

**Some technical context:** 

Previously we were using the [`followButtons.js`](https://github.com/forem/forem/blob/b423f92659d569eb69951610b67ea2b0d8ffc611/app/javascript/packs/followButtons.js#L1) to toggle the state of the following of tags on the /tags page. In fact, `followButtons.js` is essentially a JavaScript component that is used to manage **all** the followButtons across the application. Whilst this may have worked in the past, we've now introduced a "Hide" button which has affects on the "follow" button, hence causing the the /tags page follow button now deviate form this standard pattern. Trying to retrofit this pattern into the existing `followButtons.js` does not seem feasible. You can find my [very draft, very ugly attempt](https://github.com/forem/forem/pull/19921).

In addition, `followButtons.js` does a lot of manual DOM manipulation for changing the state of the buttons based on the current state. This seems like a better use case to be handled by Preact rather than doing the DOm manipulation by hand in vanilla JS. Hence, you'll notice that I created a Preact component for the buttons.

_In the future: _
It's a bit weird to be in this intermediate state where most of the Follow buttons are being handled by `followButtons.js` but then the /tags page has its own mechanism to handle these buttons. I'd like for us to improve on followButtons and turn most of that manual code into a Preact component that will manage the state of the DOM more easily and make it more readable and easy to follow. I'll do a further write-up and chat more about it in our upcoming snack n study with the team.